### ISSUE TITLE
[vue/no-useless-template-attributes] ref on v-for template is not useless

### ISSUE THREAD

**Issue by antoine-roth** [vuejs/eslint-plugin-vue#2264]
**What rule do you want to change?**
vue/no-useless-template-attributes

**Does this change cause the rule to produce more or fewer warnings?**
Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behaviour I think

**Please provide some example code that this change will affect:**

```vue
<template>
    <div>
        <template
            v-for="item in items"
            :key="item.id"
            :ref="element => setItemRef(element, item.id)"
        >
            <div>Some content</div>
        </template>
    </div>
</template>

<script setup lang="ts">
    import { ref } from "vue";

    const props = defineProps<{
        items: { id: string }[];
    }>();

	const ItemsRefs = ref<Record<string, any>>({});


    function setItemRef(element: any, itemId: string) {
        filtersRefs.value[filterId] = element;
    }
</script> 
```

**What does the rule currently do for this code?**
The rule currently flags the `ref` attribute as useless when it's not. The ref here is really useful, it's not associated to the template (that does not exist in the DOM), but to the items of the v-for

In my case I'm using a [function ref](https://vuejs.org/guide/essentials/template-refs.html#function-refs).

**What will the rule do after it's changed?**

Not flag `ref` as useless

**Additional context**

**Comment by antoine-roth** [comment]
After a second check I realized that this was not working (when it's working with a div instead of the template) 🤦 
sorry for the useless noise...
